portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,83 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import java.util.Arrays;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private Comment userComment;
+    private final ArrayList<String> comments = new ArrayList<>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+    
+    // Tutorial Steps Below:
+    // ArrayList<String> messages = new ArrayList<String>();
+    // messages.add("Alpha");
+    // messages.add("Beta");
+    // messages.add("Charlie");
+    
+    // String json = convertToJsonUsingGson(messages);
+
+    // response.setContentType("application/json");
+    // response.getWriter().println(json);
+
+    //response.setContentType("text/html;");
+    //response.getWriter().println("Hello Andrew");
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
A comment here explaining what a post request is supposed to do is good

Reply:
Completed!